### PR TITLE
comgt: pass only sensible manufacturer values (take 2)

### PR DESCRIPTION
**Pull request by samunders-core**
Duplicate of [#859](https://github.com/openwrt/openwrt/pull/859), I was unable to reopen.
My Huawei E3372h-153 on random manages to produce rssi instead of manufacturer name:
```
root@LEDE:~# gcom -d /dev/ttyUSB0 -s /etc/gcom/getcardinfo.gcom  | awk 'NF && $0
 !~ /AT\+CGMI/ { sub(/\+CGMI: /,""); print tolower($1); exit; }'
^rssi:31
root@LEDE:~#
```
This patch passes through only supported values, otherwise preserving original functionality. Awk's exit code made to mimic grep

### CODE REVIEW

**Comment by adrianschmutzler** [comment]
This needs a rebase due to the recent changes.
@jow- Can you have a look here, this also touches the manufacturer in comgt

File: package/network/utils/comgt/files/ncm.sh [hunk 1]
```diff
@@ -73,14 +73,15 @@ proto_ncm_setup() {
 
 	[ -n "$delay" ] && sleep "$delay"
 
-	manufacturer=`gcom -d "$device" -s /etc/gcom/getcardinfo.gcom | awk 'NF && $0 !~ /AT\+CGMI/ { sub(/\+CGMI: /,""); print tolower($1); exit; }'`
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_get_keys supported
+	manufacturer=`gcom -d "$device" -s /etc/gcom/getcardinfo.gcom | awk 'index("'"$supported"'",$1){f=1;print tolower($1);exit} END{exit 1-f}'`
```
Comment (adschm) [inline package/network/utils/comgt/files/ncm.sh]: Please use $() syntax instead of backticks.

**Comment by adschm** [comment]
This needs to be rebased. Please also add a commit message for all commits, bump PKG_RELEASE in the package's Makefile and remove the duplicate and merge commits.

**Comment by samunders-core** [comment]
Rebased, hopefully i understood/addressed your comments correctly. BR, Lubos

**Comment by grzesuav** [comment]
would be really good to get it merge, I was also fighting a bit with this

**Comment by adschm** [comment]
This still has formal issues. You need a proper commit title and commit message.

https://openwrt.org/submitting-patches

**Comment by samunders-core** [comment]
I messed up once more :-(

**Pull request closed without merging**